pkg/nsx/services/securitypolicy/compare.go

```diff
@@ -1,4 +1,4 @@
-package services
+package securitypolicy
```

Comment:
I'm not sure whether re-grouping the package `services` is a good idea. In the future, it's possible that multiple reconcilers need the same comparison function. For example, the subnet reconciler and subnetset reconciler may both need a function named `subnetCompare`, then how to handle this kind of cases?

Reply:
As discussed, we could import other module's func temporarily.

Reply:
For func
func (service *SecurityPolicyService) securityPolicyCompare(sp1 *model.SecurityPolicy, sp2 *model.SecurityPolicy) *model.SecurityPolicy {
}
It's better to use  like
func (service *SecurityPolicyService) securityPolicyCompare(existingSecurityPolicy *model.SecurityPolicy, securityPolicy *model.SecurityPolicy{}
It's good to align with other funcs for input arguments naming

Reply:
please change this func as well